tests/handlers/detector_view_test.py

```diff
@@ -1144,3 +1144,148 @@ def test_unchanged_roi_resend_unnecessarily_resets_cumulative(
+        toa_values = [100, 200, 300, 400, 500, 600, 700, 800]
+
+        events = DetectorEvents(
+            pixel_id=pixel_ids,
+            time_of_arrival=toa_values,
+            unit='ns',
+        )
+        grouper = GroupIntoPixels(detector_number=detector_number)
+        grouper.add(0, events)
+        detector_events = grouper.get()
+
+        view.accumulate({'detector': detector_events}, start_time=1000, end_time=2000)
+        result = view.finalize()
+
+        assert result['counts_total'].value == 8
+        # TOA range [200, 600) should include events at 200, 300, 400, 500
+        assert result['counts_in_toa_range'].value == 4
+
+    def test_counts_reset_after_finalize(
+        self,
+        mock_rolling_view: RollingDetectorView,
+        sample_detector_events: sc.DataArray,
+    ) -> None:
+        """Test that counts are reset after each finalize call."""
+        params = DetectorViewParams()
+        view = DetectorView(params=params, detector_view=mock_rolling_view)
+
+        # First accumulation
+        view.accumulate(
+            {'detector': sample_detector_events}, start_time=1000, end_time=2000
+        )
+        result1 = view.finalize()
+        assert result1['counts_total'].value == TOTAL_SAMPLE_EVENTS
+
+        # Second accumulation - counts should be fresh, not cumulative
+        view.accumulate(
+            {'detector': sample_detector_events}, start_time=2000, end_time=3000
+        )
+        result2 = view.finalize()
+        assert result2['counts_total'].value == TOTAL_SAMPLE_EVENTS
+
+    def test_counts_accumulate_within_finalize_period(
+        self,
+        mock_rolling_view: RollingDetectorView,
+        sample_detector_events: sc.DataArray,
+    ) -> None:
+        """Test that multiple accumulate calls sum counts before finalize."""
+        params = DetectorViewParams()
+        view = DetectorView(params=params, detector_view=mock_rolling_view)
+
+        # Two accumulate calls before finalize
+        view.accumulate(
+            {'detector': sample_detector_events}, start_time=1000, end_time=2000
+        )
+        view.accumulate(
+            {'detector': sample_detector_events}, start_time=2000, end_time=3000
+        )
+        result = view.finalize()
+
+        # Should have sum of both accumulations
+        assert result['counts_total'].value == 2 * TOTAL_SAMPLE_EVENTS
+
+    def test_counts_reset_by_clear(
+        self,
+        mock_rolling_view: RollingDetectorView,
+        sample_detector_events: sc.DataArray,
+    ) -> None:
+        """Test that clear() resets accumulated counts."""
+        params = DetectorViewParams()
+        view = DetectorView(params=params, detector_view=mock_rolling_view)
+
+        # Accumulate some events (don't finalize yet)
+        view.accumulate(
+            {'detector': sample_detector_events}, start_time=1000, end_time=2000
+        )
+
+        # Clear should reset counts
+        view.clear()
+
```

Comment:
Could check here that the counts are 0 to verify that the clear actually did what we think, and not for example cut the connection meaning that no new events came in, in which case the assert == TOTAL_SAMPLE_EVENTS below would still pass.